content/introduction/how-ipfs-works.md

```diff
@@ -0,0 +1,88 @@
+
+DAGs are a bit of a ["turtles all the way down"](https://ipfs.io/ipfs/QmXoypizjW3WknFiJnKLwHCnL72vedxjQkDDP1mXWo6uco/wiki/Turtles_all_the_way_down.html) scenario; that is, **everything** has a CID! You've got a file that has a CID. What if there are several files in a folder? That folder has a CID, and that CID contains the CIDs of the files underneath. In turn, those files are made up of blocks, and each of those blocks has a CID. You can see how a file system on your computer could just be a DAG. You can also, hopefully, how DAG trees start to form.
+
+Another important aspect of DAGs to note: if you change a small part of a DAG, the whole root will become different. Take a look at [this helpful illustration using bananas](https://media.consensys.net/ever-wonder-how-merkle-trees-work-c2f8b7100ed3) from our friends at Consensys.
+
+### 3 \\ The DHT
+
+<img alt="IPFS Stack - Routing and Exchange" src="../assets/ipfs_stack-exchange_routing.png" width="200px" />
+
+So, to recap, IPFS lets you to give CIDs to content, and link that content together by generating DAGs using IPLD. Now let's move on to the last piece: how you find and move that content around!
+
+In order to find which peer has certain content (_discovery_), you use a [_distributed hash table_](https://en.wikipedia.org/wiki/Distributed_hash_table), or DHT. **IPFS is essentially a database.** Just like key-value pairs, if you have a peer, it will let you find content, and if you have content, it will let you find you the peer where the content is stored.
+
+The <a hrefm src="https://libp2p.io/">libp2p project</a> is the part of the IPFS ecosystem that provides the DHT and handles peers connecting and talking to each other. It can be used as a tool for other distributed systems as well; just as there is "IPFS-flavored IPLD," there's also "IPFS-flavored libp2p."
+
+Once you know where your content is (ie, which peer or peers are storing the various blocks that make up the content you're after), you use the DHT **again** to find the current location of those peers (_routing_). So, in order to get to content, you use libp2p to query the DHT twice.
+
+You've discovered your content, and you've found the current location(s) of that content — now you need to connect to that content (_exchange_). libp2p again to the rescue! _[Maybe something here about bootstrapping, Bitswap, or Graphsync.]_
```

Comment:
`exchange` is more about locating, fetching and advertising content in the network, not `connecting to content`.

Reply:
@olizilla has a TODO for this paragraph. I'm gonna wait to rework, but _noted_.